### ISSUE TITLE
`Gem::LoadError` When executing `glimmer run` on fresh scaffold app.

### ISSUE THREAD

**Issue by jayczech23** [AndyObtiva/glimmer-dsl-swt#23]
**Desktop (please complete the following information and make sure it matches [Glimmer Pre-Requisites](https://github.com/AndyObtiva/glimmer-dsl-swt#pre-requisites) before reporting an issue):**
 - CPU Architecture [e.g. x86_64 or AArch64]

x86_64

 - OS [e.g. Linux Ubuntu 20.04.1 LTS]:

MacOS 12.4

 - Java Version [e.g. JDK 16.0.2]:

JDK 16.0.2

 - JRuby Version [e.g. jruby-9.3.1.0]:

jruby 9.3.6.0

 - Glimmer Libraries/Versions [e.g. glimmer 2.4.1 / glimmer-dsl-swt 4.21.2.0]:

4.24.1.0

**Describe the issue**
I am having an issue when trying to execute the newly created app from `glimmer scaffold` command. However, I did notice an error that occurred during the end phase of `glimmer scaffold`. I am not sure if this has anything to do with the proceeding error. Please see error at gist below:

[Error running scaffold](https://gist.github.com/jayczech23/de894cc5701f6506be00f0921bd84b8a)

Ignoring the above error, I then attempted to run newly scaffolded app as-is with `glimmer run`, and receive the following error upon doing so:

[Error on glimmer run](https://gist.github.com/jayczech23/d6e2455a606f064ff2f1c948328f9bf5)

Where could the issue be? Could it be with the environment?

**Comment by AndyObtiva** [comment]
Interesting.

I tested scaffolding successfully with v[4.24.1.0](https://rubygems.org/gems/glimmer-dsl-swt/versions/4.24.1.0) when I released it on July 22, 2022. In fact, I still have the scaffolded app on my machine, and I was able to run `glimmer run` from it just fine.

I just scaffolded a new app with v[4.24.1.0](https://rubygems.org/gems/glimmer-dsl-swt/versions/4.24.1.0), and I got the error you mentioned:

`LoadError: no such file to load -- rspec/core`

This seems to be caused by some prerequisite gems getting updated, and breaking the expectations of the glimmer-dsl-swt gem somehow. `jar-dependencies` 0.4.2 definitely seems like an offender. I wonder if there is a bug with the gem. I couldn't know for sure without spending more time digging into the problem.

The outcome of this is that scaffolding got interrupted, so development-related functionality is not fully scaffolded. However, the application code is fully scaffolded.

One workaroud for running the app that worked for me is to follow these steps:

1) Edit `Gemfile` and disable all `:development` gems, and then run `bundle`. That will get rid of the `jar-dependencies` gem dependency. The file should look like this at the end:

```ruby
# frozen_string_literal: true

source 'https://rubygems.org'

git_source(:github) {|repo_name| "https://github.com/#{repo_name}" }

gem 'glimmer-dsl-swt', '~> 4.24.1.0'

# group :development do
#   gem 'rspec', '~> 3.5.0'
#   gem 'juwelier', '2.4.9'
#   gem 'warbler', '2.0.5'
#   gem 'simplecov', '>= 0'
# end
```

2) Run the app via the scaffolded `bin/app_name` script instead of the convenience `glimmer run` command. For example, I run `bin/snowboard_utah` (name of my app), and I get this screen. 

![Screen Shot 2022-08-04 at 10 57 25 PM](https://user-images.githubusercontent.com/23052/182992087-81dd163a-cc4a-4b32-8a47-99aa5c32cff9.png)

I'm keeping this issue open till it is fully resolved, but you have a workaround in the meantime to avoid being stuck (if you cannot wait for a fix, perhaps checking out older major versions of glimmer-dsl-swt and jruby might work.)

**Comment by AndyObtiva** [comment]
I just updated my previous response with a better workaround solution that enables running `glimmer run` successfully as well as `glimmer "package[dmg]"`. Please re-read if you read my response before this latest update.

**Comment by AndyObtiva** [comment]
Good news!

The [`jar-dependencies`](https://rubygems.org/gems/jar-dependencies) gem version 0.4.2 definitely turned out to be the culprit. It was only [released yesterday](https://rubygems.org/gems/jar-dependencies/versions/0.4.2).

I just released a [v4.24.1.1 patch update](https://rubygems.org/gems/glimmer-dsl-swt/versions/4.24.1.1) for `glimmer-dsl-swt` that freezes the `jar-dependencies` gem version to 0.4.1 to avoid the issue altogether.

It now scaffolds successfully again all the way to launching the application.

For example:

```
% glimmer "scaffold[snowboard_utah]"
Fetching kamelcase-0.0.2.gem
Fetching nokogiri-1.13.8-java.gem
Fetching highline-2.0.3.gem
Fetching multi_xml-0.6.0.gem
Fetching semver2-3.4.2.gem
Fetching multi_json-1.15.0.gem
Fetching rack-2.2.4.gem
Fetching jwt-2.4.1.gem
Fetching ruby2_keywords-0.0.5.gem
Fetching faraday-retry-1.0.3.gem
Fetching faraday-rack-1.0.0.gem
Fetching faraday-patron-1.0.0.gem
Fetching faraday-net_http_persistent-1.2.0.gem
Fetching faraday-net_http-1.0.1.gem
Fetching multipart-post-2.2.3.gem
Fetching faraday-multipart-1.0.4.gem
Fetching faraday-httpclient-1.0.1.gem
Fetching faraday-excon-1.1.0.gem
Fetching faraday-em_synchrony-1.0.0.gem
Fetching faraday-em_http-1.0.0.gem
Fetching faraday-1.10.0.gem
Fetching oauth2-1.4.10.gem
Fetching hashie-3.6.0.gem
Fetching thread_safe-0.3.6-java.gem
Fetching descendants_tracker-0.0.4.gem
Fetching public_suffix-4.0.7.gem
Fetching addressable-2.8.0.gem
Fetching github_api-0.19.0.gem
Fetching rchardet-1.8.0.gem
Fetching git-1.11.0.gem
Fetching builder-3.2.4.gem
Fetching juwelier-2.4.9.gem
Successfully installed semver2-3.4.2
Successfully installed nokogiri-1.13.8-java
Successfully installed kamelcase-0.0.2
Successfully installed highline-2.0.3
Successfully installed rack-2.2.4
Successfully installed multi_xml-0.6.0
Successfully installed multi_json-1.15.0
Successfully installed jwt-2.4.1
Successfully installed ruby2_keywords-0.0.5
Successfully installed faraday-retry-1.0.3
Successfully installed faraday-rack-1.0.0
Successfully installed faraday-patron-1.0.0
Successfully installed faraday-net_http_persistent-1.2.0
Successfully installed faraday-net_http-1.0.1
Successfully installed multipart-post-2.2.3
Successfully installed faraday-multipart-1.0.4
Successfully installed faraday-httpclient-1.0.1
Successfully installed faraday-excon-1.1.0
Successfully installed faraday-em_synchrony-1.0.0
Successfully installed faraday-em_http-1.0.0
Successfully installed faraday-1.10.0

You have installed oauth2 version 1.4.10, which is EOL.
No further support is anticipated for the 1.4.x series.

OAuth2 version 2 is released.
There are BREAKING changes, but most will not encounter them, and upgrading should be easy!

Please see:
• https://github.com/oauth-xx/oauth2#what-is-new-for-v20
• https://github.com/oauth-xx/oauth2/blob/master/CHANGELOG.md

Please upgrade, report issues, and support the project! Thanks, |7eter l-|. l3oling

Successfully installed oauth2-1.4.10
Successfully installed hashie-3.6.0
Successfully installed thread_safe-0.3.6-java
Successfully installed descendants_tracker-0.0.4
Successfully installed public_suffix-4.0.7
Successfully installed addressable-2.8.0
Successfully installed github_api-0.19.0
Successfully installed rchardet-1.8.0
Successfully installed git-1.11.0
Successfully installed builder-3.2.4
Successfully installed juwelier-2.4.9
32 gems installed
	create	.gitignore
	create	Rakefile
	create	Gemfile
	create	LICENSE.txt
	create	README.markdown
	create	.document
	create	lib
	create	lib/snowboard_utah.rb
	create	.rspec
Juwelier has prepared your gem in ./snowboard_utah
Created snowboard_utah/.gitignore
Created snowboard_utah/.ruby-version
Created snowboard_utah/.ruby-gemset
Created snowboard_utah/VERSION
Created snowboard_utah/LICENSE.txt
Created snowboard_utah/Gemfile
Created snowboard_utah/Rakefile
Created snowboard_utah/app/snowboard_utah.rb
Created snowboard_utah/app/snowboard_utah/view/app_view.rb
Created snowboard_utah/icons/windows/Snowboard Utah.ico
Created snowboard_utah/icons/macosx/Snowboard Utah.icns
Created snowboard_utah/icons/linux/Snowboard Utah.png
Created snowboard_utah/app/snowboard_utah/launch.rb
Created snowboard_utah/bin/snowboard_utah
Fetching gem metadata from https://rubygems.org/.......
Resolving dependencies...................
Using rake 13.0.6
Using public_suffix 4.0.7
Using addressable 2.8.0
Using array_include_methods 1.4.0
Using awesome_print 1.9.2
Using builder 3.2.4
Using bundler 2.3.17
Using thread_safe 0.3.6 (java)
Using concurrent-ruby 1.1.10
Fetching diff-lcs 1.5.0
Fetching docile 1.4.0
Using facets 3.1.0
Using faraday-em_http 1.0.0
Using faraday-em_synchrony 1.0.0
Using faraday-excon 1.1.0
Using faraday-httpclient 1.0.1
Using multipart-post 2.2.3
Using faraday-multipart 1.0.4
Using faraday-net_http 1.0.1
Using faraday-net_http_persistent 1.2.0
Using faraday-patron 1.0.0
Using faraday-rack 1.0.0
Using faraday-retry 1.0.3
Using ruby2_keywords 0.0.5
Using rchardet 1.8.0
Using faraday 1.10.0
Using hashie 3.6.0
Using jwt 2.4.1
Using multi_xml 0.6.0
Using multi_json 1.15.0
Using glimmer 2.7.3
Using rack 2.2.4
Using jruby-win32ole 0.8.5
Using nested_inherited_jruby_include_package 0.3.0
Using puts_debuggerer 0.13.5
Using os 1.1.4
Using tty-cursor 0.7.1
Using tty-color 0.6.0
Using wisper 2.0.1
Using tty-screen 0.8.1
Using rouge 3.30.0
Using method_source 1.0.0
Using text-table 1.2.4
Using jar-dependencies 0.4.1
Using highline 2.0.3
Fetching jruby-rack 1.1.21
Fetching jruby-jars 9.3.6.0
Installing docile 1.4.0
Installing diff-lcs 1.5.0
Installing jruby-rack 1.1.21
Using semver2 3.4.2
Fetching psych 4.0.4 (java)
Fetching racc 1.6.0 (java)
Installing racc 1.6.0 (java)
Installing psych 4.0.4 (java)
Fetching rspec-support 3.5.0
Installing rspec-support 3.5.0
Fetching rubyzip 1.3.0
  jar dependencies for psych-4.0.4-java.gemspec . . .
Installing gem 'ruby-maven' . . .
Installing rubyzip 1.3.0
Fetching simplecov-html 0.12.3
Installing simplecov-html 0.12.3
Fetching simplecov_json_formatter 0.1.4
Installing simplecov_json_formatter 0.1.4
Using descendants_tracker 0.0.4
Using git 1.11.0
Using oauth2 1.4.10
Using pastel 0.8.0
Using tty-reader 0.9.0
Using super_module 1.4.1
Using kamelcase 0.0.2
Using nokogiri 1.13.8 (java)
Fetching rspec-core 3.5.4
Installing rspec-core 3.5.4
Fetching rspec-expectations 3.5.0
Installing rspec-expectations 3.5.0
Fetching rspec-mocks 3.5.0
Using github_api 0.19.0
Using tty-prompt 0.23.1
Fetching simplecov 0.21.2
Installing simplecov 0.21.2
Installing rspec-mocks 3.5.0
Using rake-tui 0.2.3
Using glimmer-dsl-swt 4.24.1.1
Fetching rspec 3.5.0
Installing jruby-jars 9.3.6.0
Installing rspec 3.5.0
Fetching warbler 2.0.5

using maven for the first time results in maven
downloading all its default plugin and can take time.
as those plugins get cached on disk and further execution
of maven is much faster then the first time.

Installing warbler 2.0.5
2022-08-04T23:49:07.324-04:00 [main] WARN FilenoUtil : Native subprocess control requires open access to the JDK IO subsystem
Pass '--add-opens java.base/sun.nio.ch=ALL-UNNAMED --add-opens java.base/java.io=ALL-UNNAMED' to enable.
      org.yaml:snakeyaml:1.28:compile
Fetching rdoc 6.4.0
Installing rdoc 6.4.0
Using juwelier 2.4.9
Bundle complete! 6 Gemfile dependencies, 74 gems now installed.
Use `bundle info [gemname]` to see where a bundled gem is installed.
   exist   .rspec
  create   spec/spec_helper.rb
Created snowboard_utah/spec/spec_helper.rb
Launching Glimmer Application: ./bin/snowboard_utah
Launching Glimmer Application: /Users/andymaleh/code/glimmer-dsl-swt/tmp/snowboard_utah/app/snowboard_utah/launch.rb
```

![screenshot hello world](https://user-images.githubusercontent.com/23052/182992087-81dd163a-cc4a-4b32-8a47-99aa5c32cff9.png)

This resolves the issue completely, so I am closing it.

In any case, if you have any other questions, feel free to reply here or hit me up in [Gitter Chat](https://gitter.im/AndyObtiva/glimmer) (or open a new issue if you encounter another one).

Cheers!

**Comment by jayczech23** [comment]
Hi @AndyObtiva thanks for the prompt reply.

That worked after updating `glimmer-dsl-swt` to `4.24.1.1`. 

Thanks for the help!
